dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/parse-filters.ts

```diff
@@ -1,5 +1,46 @@
-import { PremiumPaymentsFilters } from '@ukef/dtfs2-common';
+import { Currency, CURRENCY_REGEX, isPaymentReferenceOverFiftyCharacters, PaymentDetailsFilters, PremiumPaymentsFilters } from '@ukef/dtfs2-common';
 import { REGEX } from '../../../../../constants';
+import { ValidatedPaymentDetailsFilters } from '../../../../../types/utilisation-reports';
+
+/**
+ * Parses the provided payment details tab filters to test validity.
+ * @param paymentDetailsFilters - The filters to be applied to the fee record payment data
+ * @param paymentDetailsFilters.facilityId - The facility ID filter
+ * @param paymentDetailsFilters.paymentCurrency - The payment currency filter
+ * @param paymentDetailsFilters.paymentReference - The payment reference filter
+ * @returns The parsed filters for the payment details tab
+ */
+export const parsePaymentDetailsFilters = (paymentDetailsFilters?: PaymentDetailsFilters): ValidatedPaymentDetailsFilters => {
+  if (!paymentDetailsFilters) {
+    return {};
+  }
+
+  const { facilityId, paymentCurrency, paymentReference } = paymentDetailsFilters;
+
+  let facilityIdFilter;
+
+  if (facilityId && REGEX.UKEF_PARTIAL_FACILITY_ID_REGEX.test(facilityId)) {
+    facilityIdFilter = facilityId;
+  }
+
+  let paymentReferenceFilter;
+
+  if (paymentReference && !isPaymentReferenceOverFiftyCharacters(paymentReference) && paymentReference.length >= 4) {
```

Comment:
`!isPaymentReferenceOverFiftyCharacters(paymentReference) && paymentReference.length >= 4` could this not be a function itself?

Reply:
Pulled this out into its own function thanks